python/fledge/services/core/service_registry/monitor.py

```diff
@@ -145,13 +150,48 @@ async def _monitor_loop(self):
                 if check_count[service_record._id] > self._max_attempts:
                     ServiceRegistry.mark_as_failed(service_record._id)
                     check_count[service_record._id] = 0
+                    if self._support_bundle_config['auto_support_bundle']['value'] == 'true':
+                        self._logger.info("Service %s failed, creating automated support bundle",
+                                          service_record._name)
+                        asyncio.create_task(self.create_automated_support_bundle(service_record._name))
                     try:
                         audit = AuditLogger(connect.get_storage_async())
                         await audit.failure('SRVFL', {'name':service_record._name})
                     except Exception as ex:
                         self._logger.info("Failed to audit service failure %s", str(ex))
             await self._sleep(self._sleep_interval)
 
+    async def create_automated_support_bundle(self, service_name):
+        """Create support bundle asynchronously when service fails"""
+        try:
+            from fledge.services.core.support import SupportBuilder
+            from fledge.common.common import _FLEDGE_DATA
+            support_dir = _FLEDGE_DATA + "/support" if _FLEDGE_DATA else _FLEDGE_ROOT + "/data/support"
+            builder = SupportBuilder(support_dir, self._support_bundle_config)
+            bundle_name = await builder.build(service_name)
+            # Raise alert about support bundle creation
+            await self.raise_support_bundle_alert(service_name, bundle_name)
+            
+            self._logger.info("Support bundle created: %s for failed service: %s",
+                            bundle_name, service_name)
+        except Exception as ex:
+            self._logger.error(ex, "Failed to create support bundle for {}".format(service_name))
```

Comment:
Avoid mixing string formatting styles; choose one between % and .format() and apply it consistently.
The suggested method for string formatting is to utilize `f-strings`.